core/src/main/java/org/apache/iceberg/GenericTrackedFile.java

```diff
@@ -0,0 +1,773 @@
+
+  // Tracking metadata
+  private String manifestLocation = null;
+  private TrackingInfo.Status status = null;
+  private Long snapshotId = null;
+  private Long sequenceNumber = null;
+  private Long fileSequenceNumber = null;
+  private Long firstRowId = null;
+  private Long position = null;
+
+  // Common file metadata
+  private FileContent contentType = FileContent.DATA;
+  private String location = null;
+  private FileFormat fileFormat = null;
+  private long recordCount = -1L;
+  private Long fileSizeInBytes = null;
+
+  // Deletion vector fields
+  private Long deletionVectorOffset = null;
+  private Long deletionVectorSizeInBytes = null;
+  private byte[] deletionVectorInlineContent = null;
+
+  // Partition and ordering
+  private int partitionSpecId = -1;
+  private Integer sortOrderId = null;
+
+  // Optional metadata
+  private byte[] keyMetadata = null;
+  private long[] splitOffsets = null;
+  private int[] equalityIds = null;
+  private String referencedFile = null;
+
+  // Manifest stats (for manifest entries)
+  private Integer addedFilesCount = null;
+  private Integer existingFilesCount = null;
+  private Integer deletedFilesCount = null;
+  private Long addedRowsCount = null;
+  private Long existingRowsCount = null;
+  private Long deletedRowsCount = null;
+  private Long minSequenceNumber = null;
+
+  // Content stats placeholder (TODO: implement ContentStats)
+  private Object contentStats = null;
+
+  // Cached schema for Avro
+  private transient Schema avroSchema = null;
+
+  // Base type that corresponds to positions for get/set
+  // All fields are made optional to support different content types
+  static final Types.StructType BASE_TYPE =
+      Types.StructType.of(
+          TrackedFile.CONTENT_TYPE,
+          TrackedFile.LOCATION,
+          TrackedFile.FILE_FORMAT,
+          TrackedFile.PARTITION_SPEC_ID,
+          TrackedFile.SORT_ORDER_ID,
+          TrackedFile.RECORD_COUNT,
+          TrackedFile.FILE_SIZE_IN_BYTES,
+          TrackedFile.KEY_METADATA,
+          TrackedFile.SPLIT_OFFSETS,
+          TrackedFile.EQUALITY_IDS,
+          TrackedFile.REFERENCED_FILE,
+          // Tracking info fields
+          TrackingInfo.STATUS.asOptional(),
+          TrackingInfo.SNAPSHOT_ID,
+          TrackingInfo.SEQUENCE_NUMBER,
+          TrackingInfo.FILE_SEQUENCE_NUMBER,
+          TrackingInfo.FIRST_ROW_ID,
+          // Deletion vector fields
+          DeletionVector.OFFSET,
+          DeletionVector.SIZE_IN_BYTES,
+          DeletionVector.INLINE_CONTENT,
+          // Manifest stats fields
+          ManifestStats.ADDED_FILES_COUNT.asOptional(),
+          ManifestStats.EXISTING_FILES_COUNT.asOptional(),
+          ManifestStats.DELETED_FILES_COUNT.asOptional(),
+          ManifestStats.ADDED_ROWS_COUNT.asOptional(),
+          ManifestStats.EXISTING_ROWS_COUNT.asOptional(),
+          ManifestStats.DELETED_ROWS_COUNT.asOptional(),
+          ManifestStats.MIN_SEQUENCE_NUMBER.asOptional());
```

Comment:
Each of the sub-structs should be its own struct type (like `ManifestStatsStruct` and `TrackingInfoStruct`) that is read and written the same way that this class is. The `TrackedFile` schema should embed each sub struct as a field.